kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/bootstrap/RandomBootstrapSelectionStrategy.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.bootstrap;
+
+import java.util.List;
+
+import io.kroxylicious.proxy.service.HostPort;
+
+/**
+ * Strategy which chooses a random upstream target as the bootstrap server.
+ */
+public class RandomBootstrapSelectionStrategy implements BootstrapSelectionStrategy {
+
+    @Override
+    public HostPort apply(List<HostPort> hostPorts) {
+        final int choice = (int) (Math.random() * hostPorts.size());
```

Comment:
Does this always produce a value within the size of the list? 

I think one needs something more like https://www.baeldung.com/java-generating-random-numbers-in-range#intstreamiterate 

Or using a Random instance and specifying an [upper bound ](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Random.html#nextInt(int))

Reply:
Your right about the range, not sure what I was thinking about earlier. 

I don't think changing to an instance is needed.